chainer/__init__.py

```diff
@@ -89,6 +93,45 @@ def get_function_hooks():
     return ret
 
 
+def _load_array_types():
+    global _array_types
+    global _cpu_array_types
+    if _array_types is None:
+        _array_types = [numpy.ndarray]
+        _cpu_array_types = [numpy.ndarray]
```

Comment:
Since we are modifying globals, I feel that the above two variables should be temporary variables and that `_array_types` which is used in the condition should be set to a non-`None` value at the very end. Is it unnecessary you think?

Reply:
Ah, I see.
I thought synchronization is not needed because it's protected by GIL, but the call to `is_ideep_available()` may incur a IO wait (although it's not the case in the current implementation).